### PR TITLE
[Card#19]-Create leave feature-Dependancy on default_calendar_settings branch

### PR DESCRIPTION
**Pull request by HarshwardhanSingh**
https://trello.com/c/v9kdZOsN

Dependent on **default_calendar_settings** branch. Branched out from master.

### CODE REVIEW

File: app/controllers/leaves_controller.rb [hunk 1]
```diff
@@ -4,16 +4,32 @@ def index
   end
 
   def new
+    @leave = Leave.new
```
Comment (deepak) [inline app/controllers/leaves_controller.rb]: no need for instance variable as you are using locals for rendering

File: app/controllers/leaves_controller.rb [hunk 2]
```diff
@@ -4,16 +4,32 @@ def index
...
+    render locals:{calendars: Invite.where(user_id: current_user.id).where(is_default: true).map(&:calendar_id)}
   end
 
   def create
+    if(params[:leave_duration]=="First Half" || params[:leave_duration]=="Second Half")
```
Comment (deepak) [inline app/controllers/leaves_controller.rb]: "First Half" and "Second Half" can be a constant

**Comment by HarshwardhanSingh** [comment]
Currently we are using half day parameter to write the summary of leave like "Harshwardhan is on Half Day leave(Reason)". Are we planning to calculate like 2 half day leaves = 1 full day leave. If yes then we will need additional attribute in leaves table.

**Comment by deepak** [comment]
in that case, will need to:

1. do a lookup in the past
2. delete old event
3. create a new event

so can be a separate story

**Comment by HarshwardhanSingh** [comment]
okay

**Comment by deepak** [comment]
@HarshwardhanSingh once the comments are fixed, please merge the branch
-

**Pull request merged**